src/App.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useCallback, useRef } from 'react';
+import Controls from './components/controls';
+import produce from 'immer';
+import './App.css';
+
+const neighborLocations = [
+  [0, 1],
+  [0, -1],
+  [1, -1],
+  [-1, 1],
+  [1, 1],
+  [-1, -1],
+  [1, 0],
+  [-1, 0]
+];
+
+function App() {
+
+  const [grid, setGrid] = useState();
+  const [cols, setCols] = useState();
+  const [rows, setRows] = useState();
+  const [running, setRunning] = useState(false);
+  const [speed, setSpeed] = useState(1000)
```

Comment:
You may have planned on doing this in the future, but at some point it may be better to switch to useContext to organize all of your state in one spot.